sessions/js-array-methods/filter/index.js

```diff
@@ -0,0 +1,40 @@
+console.clear();
+
+const cards = [
+	{
+		id: '1',
+		question: 'How often can I use <header>?',
+		answer: 'As often as you like.',
+		tags: ['html', 'elements', 'basic'],
+	},
+	{
+		id: '2',
+		isBookmarked: false,
+		question: 'How often can I use <aside>?',
+		answer: 'As often as you like.',
+		tags: ['html', 'elements', 'basic'],
+	},
+	{
+		id: '3',
+		isBookmarked: true,
+		question: 'On which types can I use destructuring assignment?',
+		answer: 'On Objects and Arrays',
+		tags: ['js', 'next'],
+	},
+];
+
+// Filter: returns an array of max. same length
+// but with unaltered content
+// is called with a function that returns a 'truthy' or 'falsy' value
+const notBookmarkedCards = cards.filter(card => !card.isBookmarked);
+console.log(notBookmarkedCards);
+
+// 1: Filter the array for cards with the 'html' tag
+// HINT: use the array method "includes"
+const onlyCardsWithTagHTML = null;
+
+// 2: Filter the array for cards with 3 tags.
+const allCardsWith3Tags = null;
+
+// 3: Filter the array for cards with the 'html' and 'js' tag that are also bookmarked.
+const allCardsWithTagsHTMLAndJSThatAreBookmarked = null;
```

Comment:
Here is my solution:
https://codesandbox.io/s/adoring-tree-roeog4?file=/index.js

I was not sure about what task 3. is asking for. Is it what you expected?

Reply:
@doemser your solution doesn't make sense:

```js
(card.tags.includes("html") || card.tags.includes("html")) 
```

I did not create the original exercise. @F-Kirchhoff do you remember who created this exercise.

I am also unsure what the question is asking for.

Here's how I understand it:

Create an array of all cards that have "html" AND "js" that are bookmarked.

Example: 

```js
const allCardsWithTagsHTMLAndJSThatAreBookmarked = cards.filter((card) => {
  return (
    (card.tags.includes("html") && card.tags.includes("js")) && card.isBookmarked
  );
});
```

This will return an empty array since none of the cards have "html" AND "js" 🤷

Reply:
> @doemser your solution doesn't make sense: :trollface:
> 
> ```js
> (card.tags.includes("html") || card.tags.includes("html")) 
> ```
> 

Oh, hahaha :D What happened there?! This is what I meant

```js
(card.tags.includes("html") || card.tags.includes("js")) &&
    card.isBookmarked
```

Reply:
Here Gregor is right, the solution is an empty array, therefore lets change the exercise as follows:
```suggestion
// 1: Filter the array for cards with the 'html' tag
// HINT: use the array method "includes"
const onlyCardsWithTagHTML = null;

// 2: Filter the array for cards with 3 tags.
const allCardsWith3Tags = null;

// 3: Filter the array for cards with the 'html' or 'js' tag that are also bookmarked.
const allCardsWithTagsHTMLOrJSThatAreBookmarked = null;
```